### PR TITLE
Update to Version 2 of the dataset

### PR DESCRIPTION
**Pull request by Olivierj23**
Hi, my name is Olivier Lafontant-Joseph. I am an undergraduate student working with Jonathan Zheng on the IUPAC pKa dataset.

I have added version 2 of the IUPAC high-confidence dataset. 

The following changes have been made:

- cis/trans smiles have been added where possible
- additional SMILES strings have been added
- solvent column has been added with parsed solvents from remarks
- InChI strings have been added in a column
- standardization of temperature values to floats where possible
- marked uncertain entries (such entries have "?" behind pKa type and acidity label) based on non-standard pKa types/ pKa value  and when pKa types were ambiguous for amphoteric/polyprotic molecules
- fixed acidity labels for amino acids
- replaced entry #s with unique IDs (source + entry number)
- Parsed more pressure values from remarks

Added serjeant scans to ref folder

README and the digitization report pdf will be updated soon.

Please let me know if you have any questions about these changes.

### CODE REVIEW

File: .DS_Store
Comment (jonwzheng) [inline .DS_Store:1]: could we remove this from the PR?

**Comment by jonwzheng** [comment]
To expand on some of the changes we made:
> **additional SMILES strings**

In version 1.0, we had many entries with SMILES strings that could not be verified using computational approaches. Another subset had "conflicting" SMILES strings, i.e. different software produced multiple different translations. And yet another subset included entries with fewer than 2 agreeing translations, i.e. only 1 software was able to translate the name, so the structure needed to be checked. 

I checked or manually constructed SMILES strings for those several thousand problematic entries. Most of the original mistakes were systematic, so they could be corrected with fairly high confidence. Also, the reference books often included "clues" to the correct structure, such as through diagrams or molecular formulae. So, I have high confidence that the vast majority of these new entries are named consistently with the authors' intentions.

> **uncertain entries for amphoteric or polyprotic molecules**

Some entries have multiple pK's listed corresponding to different ionizable centers, like pk1 = 2.1, pk2 = 3.6, pk3 = 5.7, pk4 = 8.4. These can either be polyprotic (i.e. all are acidic dissociations) or amphoteric (i.e. some are basic dissociations, some are acidic). In the Perrin works, there is clueing based on the order of the pKa values that helps make this determination: if the values are in strictly decreasing order, then they are all basic dissociations, and we can safely label them as pKAH1, pKAH2, etc. But if they're in strictly increasing order, then they are amphoteric. Serjeant has no such clueing, so all entries with multiple acidities can be either amphoteric or polyprotic. Unfortunately, for amphoteric molecules, we cannot determine (without examining each structure) which values are acidic and which are basic. So we can't assign their acidity types. Therefore, we added question marks to those entries with this issue. All in all, this is something like 6,000 entries whose acidity types still need to be clarified.  The only exception to this is if two pKa values are reported. In such cases, it is unambiguous that the lowest dissociation value is the basic dissociation and the higher one is acidic.

One could try to use computational information about the molecular structure to infer amphoteric/polyprotic; this is something we're looking at doing in a future release.

> **standardization of temperature values to floats where possible**

For modeling purposes, it is useful if all temperatures are numbers rather than having a few of them as strings. We made just a few modifications: "room temperature" now refers to 25 degrees C, and temperature ranges like "18-22" take the lower temperature. If desired, the original temperatures are still in the dataset as a separate column.

> **solvent column has been added with parsed solvents from remarks**

Some measurements were made in dilute solutions, and it may be useful for modelers to isolate these from pure aqueous measurements. 

****

A big thank-you to @Olivierj23 for effecting many of the changes described herein. Please take a look and let us know if there is anything you would like to discuss more before merging in & updating as a new major release.

File: .DS_Store
Comment (LeahRaeMcEwen) [reply .DS_Store:1]: I would like to have our PubChem colleagues take a look at the SMILES and InChI additions as they did for version 1. I will be following up with them tomorrow and will let you know.

Comment (jonwzheng) [reply .DS_Store:1]: Hi - just following up to see if there's any updates to this.

Comment (Olivierj23) [reply .DS_Store:1]: I thought I removed .DS_Store in the last commit. Is it still there?

Comment (jonwzheng) [reply .DS_Store:1]: Hi Olivier - `.DS_Store` is removed, we're all set there. My comment was in response to Leah's comment on this review.

Comment (stuchalk) [reply .DS_Store:1]: The .DS_Store file is an invisible file from macos and will come back as you use files in a folder.  I would add it to the .gitignore file as a global ignore.  See https://stackoverflow.com/questions/18393498/gitignore-all-the-ds-store-files-in-every-folder-and-subfolder.  You also might need to delete this file on github after the .gitignore file is committed and pushed to the repo.

**Comment by jonwzheng** [comment]
We have internally reviewed the changes and think it would be most beneficial to push these changes as soon as possible, due to the very large number of fixes this version includes.

**Pull request merged**